pkg/R/types.R

```diff
@@ -0,0 +1,47 @@
+# Utility functions for handling Spark DataTypes.
+
+# Handler for StructType
+structType <- function(st) {
+  obj <- structure(new.env(parent = emptyenv()), class = "structType")
+  obj$jobj <- st
+  obj$fields <- lapply(SparkR:::callJMethod(st, "fields"), structField)
+  obj
+}
+
+#' Print a Spark StructType.
+#'
+#' This function prints the contents of a StructType returned from the
+#' SparkR JVM backend.
+#'
+#' @param x A StructType object
+#' @param ... further arguments passed to or from other methods
+print.structType <- function(x, ...) {
+  fieldsList <- lapply(x$fields, function(i) i$print)
```

Comment:
Minor style nit - I think we put `{}` even for one one line `function` definitions.